### PR TITLE
added environment variable to optionally set do_work frequency

### PR DESCRIPTION
**Pull request by JanZachmann**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,9 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.11.7] Q4 2023
```
Comment (mlilien) [inline CHANGELOG.md:8]: Q1 2024

File: src/client/mod.rs [hunk 2]
```diff
@@ -119,6 +120,8 @@ mod twin;
 
 static mut IOTHUB_INIT_RESULT: i32 = -1;
 static IOTHUB_INIT_ONCE: Once = Once::new();
+static DO_WORK_FREQUENCY_IN_MS: &str = "DO_WORK_FREQUENCY_IN_MS";
```
Comment (mlilien) [inline src/client/mod.rs:123]: i would prefer a azure sdk context. e.g. when setting it in a service file, it's not clear that this only refers to the azure sdk worker loop-
e.g. `AZURE_SDK_DO_WORK_FREQUENCY_IN_MS`

Comment (JanZachmann) [reply src/client/mod.rs]: done

File: CHANGELOG.md (hunk 1 above)
Comment (JanZachmann) [reply CHANGELOG.md]: done

**Review by mlilien — APPROVED** [review]

**Pull request merged**